prow/plugins/trigger/ic.go

```diff
@@ -65,6 +67,24 @@ func handleIC(c client, ic github.IssueCommentEvent) error {
 		return err
 	}
 
+	if shouldRetestFailed {
+		combinedStatus, err := c.GitHubClient.GetCombinedStatus(org, repo, pr.Head.SHA)
+		if err != nil {
+			return err
+		}
+		skipContexts := make(map[string]bool) // these succeeded or are running
+		runContexts := make(map[string]bool)  // these failed and should be re-run
+		for _, status := range combinedStatus.Statuses {
+			if status.State == "success" || status.State == "pending" {
+				skipContexts[status.Context] = true
+			} else if status.State == "error" || status.State == "failure" {
+				runContexts[status.Context] = true
+			}
+		}
+		requestedJobs = append(requestedJobs, c.Config.RetestPresubmits(ic.Repo.FullName, skipContexts, runContexts)...)
```

Comment:
That's certainly not great, but also probably not the end of the world. Would it be too much of a hassle to fix that here?